owncloudApp/src/main/java/com/owncloud/android/sharees/presentation/UsersAndGroupsSearchProvider.kt

```diff
@@ -0,0 +1,315 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David A. Velasco
+ * @author Juan Carlos González Cabrero
+ * @author David González Verdugo
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.sharees.presentation
+
+import android.app.SearchManager
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.content.UriMatcher
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import android.os.Handler
+import android.os.Looper
+import android.provider.BaseColumns
+import android.widget.Toast
+import com.owncloud.android.R
+import com.owncloud.android.authentication.AccountUtils
+import com.owncloud.android.datamodel.FileDataStorageManager
+import com.owncloud.android.lib.common.OwnCloudAccount
+import com.owncloud.android.lib.common.OwnCloudClientManagerFactory
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.shares.GetRemoteShareesOperation
+import com.owncloud.android.lib.resources.shares.ShareType
+import com.owncloud.android.operations.common.OperationType
+import com.owncloud.android.sharees.data.datasources.OCRemoteShareesDataSource
+import com.owncloud.android.sharees.domain.OCShareeRepository
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import com.owncloud.android.vo.Resource
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.ArrayList
+import java.util.HashMap
+
+/**
+ * Content provider for search suggestions, to search for users and groups existing in an ownCloud server.
+ */
+class UsersAndGroupsSearchProvider : ContentProvider() {
+    private var uriMatcher: UriMatcher? = null
```

Comment:
instead of being nullable, use lateinit var ;-)

Reply:
Done